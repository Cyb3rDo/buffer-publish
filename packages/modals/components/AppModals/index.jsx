import React from 'react';
import PropTypes from 'prop-types';

import UpgradeModal from '@bufferapp/publish-upgrade-modal';
import WelcomeModal from '@bufferapp/publish-welcome-modal';

const AppModals = ({ showUpgradeModal, showWelcomeModal }) => (
  <React.Fragment>
    {showUpgradeModal && <UpgradeModal />}
    {showWelcomeModal && <WelcomeModal />}
  </React.Fragment>
);

AppModals.propTypes = {
  showUpgradeModal: PropTypes.bool.isRequired,
  showWelcomeModal: PropTypes.bool.isRequired,
};

export default AppModals;
