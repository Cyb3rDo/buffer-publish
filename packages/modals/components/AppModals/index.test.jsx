import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';

import AppModalsConnected from '../../index';
import AppModals from './index';

const fakeStore = state => ({
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => ({ ...state }),
});

const renderer = new ShallowRenderer();

describe('Modals', () => {
  it('renders nothing when no modals are visible', () => {
    const visibleModals = { showUpgradeModal: false };
    const rendered = renderer.render(
      <AppModals {...visibleModals} />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders upgrade modal', () => {
    const visibleModals = { showUpgradeModal: true };
    const rendered = renderer.render(
      <AppModals {...visibleModals} />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders a connected upgrade modal', () => {
    const visibleModals = { showUpgradeModal: true };
    const rendered = renderer.render(
      <Provider store={fakeStore({ modals: visibleModals })}>
        <AppModalsConnected />
      </Provider>
    );
    expect(rendered).toMatchSnapshot();
  });
});
