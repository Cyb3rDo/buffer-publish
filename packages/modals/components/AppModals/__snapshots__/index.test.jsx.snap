// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals renders a connected upgrade modal 1`] = `<Connect(AppModals) />`;

exports[`Modals renders nothing when no modals are visible 1`] = `false`;

exports[`Modals renders upgrade modal 1`] = `<Connect(UpgradeModal) />`;
